statsmodels/base/model.py

```diff
@@ -799,20 +799,20 @@ def predict(self, exog=None, transform=True, *args, **kwargs):
 
         exog_index = exog.index if _is_using_pandas(exog, None) else None
 
-        if transform and hasattr(self.model, 'formula') and exog is not None:
+        if transform and hasattr(self.model, 'formula') and (exog is not None):
             from patsy import dmatrix
-            exog = pd.DataFrame(exog)  # user may pass series, if one predictor
-            if exog_index is None:  # user passed in a dictionary
-                exog_index = exog.index
+            if isinstance(exog, pd.Series):
+                exog = pd.DataFrame(exog)
+            orig_exog_len = len(exog)
             exog = dmatrix(self.model.data.design_info.builder,
                            exog, return_type="dataframe")
-            if len(exog) < len(exog_index):
-                # missing values, rows have been dropped
-                if exog_index is not None:
-                    exog = exog.reindex(exog_index)
+            if orig_exog_len > len(exog):
+                import warnings
+                if exog_index is None:
+                    warnings.warn('nan values have been dropped', ValueWarning)
                 else:
-                    import warnings
-                    warnings.warn("nan rows have been dropped", ValueWarning)
+                    exog = exog.reindex(exog_index)
+            exog_index = exog.index
```

Comment:
this line changes the behavior to before.
if exog_index is None, then this will create an exog_index and return a pandas Series (or DataFrame) even if it wasn't used in exog. Before this change it returned a numpy array in that case.
AFAICS (there are too many possibilities to keep things straight.)

Reply:
> Before this change it returned a numpy array in that case.

I'm not sure I agree with that. This is the current code:

https://github.com/statsmodels/statsmodels/blob/9f2ed83a2e38347c4eb85b48082a9e952d04fba2/statsmodels/base/model.py#L798-L808

When `exog` is provided, you get a `DataFrame` regardless.

Reply:
You're right.
I tried a plain dict exog example, and I get a pandas Series back.
(using master, I'm trying out a few examples before checking out this PR.)